File: apps/android/app/src/main/java/com/focusguard/ui/components/ApplicationCardRow.kt
package com.focusguard.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.focusguard.ui.theme.FocusGuardColors

data class AppUsageModel(
    val name: String,
    val usedMinutes: Int,
    val limitMinutes: Int,
    val themeColor: Color
) {
    val remainingMinutes: Int get() = (limitMinutes - usedMinutes).coerceAtLeast(0)

    val statusText: String get() = when {
        usedMinutes >= limitMinutes -> "Limit Reached"
        remainingMinutes <= 5 -> "Almost at limit"
        else -> "Active"
    }

    val statusColor: Color get() = when {
        usedMinutes >= limitMinutes -> FocusGuardColors.Danger
        remainingMinutes <= 5 -> FocusGuardColors.Warning
        else -> FocusGuardColors.Success
    }
}

@Composable
fun ApplicationCardRow(item: AppUsageModel) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(FocusGuardColors.Surface, RoundedCornerShape(12.dp))
            .border(1.dp, FocusGuardColors.Border, RoundedCornerShape(12.dp))
            .padding(14.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(
                text = item.name,
                fontSize = 15.sp,
                fontWeight = FontWeight.SemiBold,
                color = FocusGuardColors.TextPrimary
            )

            Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                Text(
                    text = "${item.usedMinutes}m / ${item.limitMinutes}m",
                    fontSize = 12.sp,
                    fontFamily = FontFamily.Monospace,
                    fontWeight = FontWeight.Bold,
                    color = FocusGuardColors.TextSecondary
                )
                Text(
                    text = "• ${item.remainingMinutes}m remaining",
                    fontSize = 12.sp,
                    color = if (item.remainingMinutes <= 5) FocusGuardColors.Warning else FocusGuardColors.TextMuted
                )
            }
        }

        Box(
            modifier = Modifier
                .background(item.statusColor.copy(alpha = 0.12f), RoundedCornerShape(6.dp))
                .border(1.dp, item.statusColor.copy(alpha = 0.3f), RoundedCornerShape(6.dp))
                .padding(horizontal = 8.dp, vertical = 4.dp)
        ) {
            Text(
                text = item.statusText,
                fontSize = 11.sp,
                fontWeight = FontWeight.Bold,
                color = item.statusColor
            )
        }
    }
}

File: apps/macos/FocusGuard/UI/ApplicationCardRow.swift
import SwiftUI

public struct AppUsageItem: Identifiable {
    public let id = UUID()
    public let name: String
    public let iconName: String
    public let category: String
    public let usedMinutes: Int
    public let limitMinutes: Int
    public let themeColor: Color

    public init(name: String, iconName: String, category: String, usedMinutes: Int, limitMinutes: Int, themeColor: Color) {
        self.name = name
        self.iconName = iconName
        self.category = category
        self.usedMinutes = usedMinutes
        self.limitMinutes = limitMinutes
        self.themeColor = themeColor
    }

    public var remainingMinutes: Int {
        return max(0, limitMinutes - usedMinutes)
    }

    public var statusText: String {
        if usedMinutes >= limitMinutes {
            return "Limit Reached"
        } else if remainingMinutes <= 5 {
            return "Almost at limit"
        } else {
            return "Active"
        }
    }

    public var statusColor: Color {
        if usedMinutes >= limitMinutes {
            return FocusGuardTheme.Colors.danger
        } else if remainingMinutes <= 5 {
            return FocusGuardTheme.Colors.warning
        } else {
            return FocusGuardTheme.Colors.success
        }
    }
}

public struct ApplicationCardRow: View {
    let item: AppUsageItem

    public init(item: AppUsageItem) {
        self.item = item
    }

    public var body: some View {
        HStack(spacing: 16) {
            // Icon
            ZStack {
                RoundedRectangle(cornerRadius: 10)
                    .fill(item.themeColor.opacity(0.16))
                    .frame(width: 42, height: 42)

                Image(systemName: item.iconName)
                    .font(.system(size: 20))
                    .foregroundColor(item.themeColor)
            }

            // App Name & Remaining Time
            VStack(alignment: .leading, spacing: 3) {
                Text(item.name)
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundColor(FocusGuardTheme.Colors.textPrimary)

                HStack(spacing: 6) {
                    Text("\(item.usedMinutes)m / \(item.limitMinutes)m")
                        .font(.system(size: 12, weight: .bold, design: .monospaced))
                        .foregroundColor(FocusGuardTheme.Colors.textSecondary)

                    Text("•")
                        .foregroundColor(FocusGuardTheme.Colors.textMuted)

                    Text("\(item.remainingMinutes)m remaining")
                        .font(.system(size: 12, weight: .medium))
                        .foregroundColor(item.remainingMinutes <= 5 ? FocusGuardTheme.Colors.warning : FocusGuardTheme.Colors.textMuted)
                }
            }

            Spacer()

            // Status Tag
            Text(item.statusText)
                .font(.system(size: 11, weight: .bold))
                .foregroundColor(item.statusColor)
                .padding(.horizontal, 10)
                .padding(.vertical, 5)
                .background(item.statusColor.opacity(0.12))
                .cornerRadius(FocusGuardTheme.Radius.small)
                .overlay(
                    RoundedRectangle(cornerRadius: FocusGuardTheme.Radius.small)
                        .stroke(item.statusColor.opacity(0.25), lineWidth: 1)
                )
        }
        .padding(14)
        .background(FocusGuardTheme.Colors.surface)
        .cornerRadius(FocusGuardTheme.Radius.medium)
        .overlay(
            RoundedRectangle(cornerRadius: FocusGuardTheme.Radius.medium)
                .stroke(FocusGuardTheme.Colors.border, lineWidth: 1)
        )
    }
}
